lib/mongoid/history/trackable.rb

```diff
@@ -325,10 +325,10 @@ def tracked_fields_for_action(action)
         #
         # @return [ Array < String > ] the base list of tracked database field names
         def tracked_fields
-          @tracked_fields ||= fields.keys.select do |field|
+          @tracked_fields ||= begin
             h = history_trackable_options
-            (h[:on] == :all || h[:on].include?(field)) && !h[:except].include?(field)
-          end - reserved_tracked_fields
+            (h[:on] == :all ? fields.keys : h[:on]) - h[:except] - reserved_tracked_fields
```

Comment:
I believe `fields.keys` is an array of `String`, something needs to normalize that into either strings or symbols. Probably strings since symbols used to leak and it's still a problem in some Ruby versions and because fields.keys is always strings.